### PR TITLE
[Sierra_Wireless]  Reboot Status - Object 10371

### PR DESCRIPTION
**Pull request by sbernard31**
This is triggered by #502.

This aims to resolves : https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/280.

IMHO, this should be part of the LWM2M specification (directly added to Device object) and should not require an extension.

But for LWM2M 1.0.x, it's too late and an new object is needed to resolve this issue.
For next version of the specification, maybe this should be considered as an addition ? (making this object obsolete in the future)

### CODE REVIEW

File: 10371.xml
Comment (sbernard31) [inline 10371.xml]: I use a short description here as it is adviced to use 134 character max.

(see https://github.com/OpenMobileAlliance/lwm2m-registry/wiki/Guidelines#description)

File: 10371.xml
Comment (sbernard31) [inline 10371.xml]: I use `Description2` for a more complete description. (I don't know if this is the idea)

Comment (jpradocueva) [reply 10371.xml]: Description 2 is used for a variety of situations. Therefore, it is ok to add further details if needed.

File: 10371.xml
Comment (akeranen) [reply 10371.xml]: If this "extends" the device object, should these be considered for a new version of the device object itself? If that's not good option, perhaps better to say "provides added functionality beyond.." or similar. And at large, maybe we (DMSE?) 
should think what kind of paradigm and design pattern we really want for this kind of extensions to the management objects.

File: 10371.xml
Comment (akeranen) [reply 10371.xml]: > Without this extension this is really hard to know if a reboot or a reset factory is a success [...]

This kind of description sounds weird in an object. Makes sense as part of the discussion here when deciding if/how this should be added, but not as long-term description in an object.

File: 10371.xml
Comment (akeranen) [inline 10371.xml]: I'd recommend full words for the name. For example "Extended Device Management".

Comment (dnav) [reply 10371.xml]: Maybe a more specific name ? Like "Reboot Status" as this Object's Resources are dealing only with the reboot and factory reset states.

File: 10371.xml
Comment (dnav) [inline 10371.xml]: The "Reboot" feature is mandatory to support for LwM2M Device but "Factory Reset" is optional. Thus I would suggest to make this Resource Optional.

File: 10371.xml
Comment (dnav) [inline 10371.xml]: The "Reboot" feature is mandatory to support for LwM2M Device but "Factory Reset" is optional. Thus I would suggest to make this Resource Optional.

File: 10371.xml
Comment (sbernard31) [reply 10371.xml]: Definitely a new version of Device Object would be better.
This Object just aims to fix that it seems to me as an LWM2M core issue that we discuss in https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/280#issuecomment-363483045

But we need this new version for LWM2M 1.0.x and I don't know if this is doable ? 
If this is possible then we don't really need this new object.

File: 10371.xml
Comment (sbernard31) [reply 10371.xml]: If we keep this object (https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504/files#r463470429) I will change it.

File: 10371.xml
Comment (sbernard31) [reply 10371.xml]: If we keep this object (https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504/files#r463470429) I will change it.

File: 10371.xml
Comment (sbernard31) [reply 10371.xml]: The idea was to explain rational of this object.
I thought that it could help people to decide if they need to implement it or not.

File: 10371.xml
Comment (sbernard31) [reply 10371.xml]: I'm not strongly attached to the name, so no problem to change it.
Maybe `Reboot Status` or `Device Status` could be OK.
Let me know which name sounds more appropriate to you ?
Then If we decide to keep this object (https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504/files#r463470429) I will change it.

**Comment by sbernard31** [comment]
(Just to let you know I will be unavailable the next 2 weeks)

File: 10371.xml
Comment (akeranen) [reply 10371.xml]: For that I’d recommend wording like “This object provides additional information on whether xxx succeeded...” etc. “really hard to know” sounds very subjective.

**Comment by seanmcilroy29** [comment]
@sbernard31 - Can you confirm if all the submitted observations have been resolved?

**Comment by sbernard31** [comment]
@seanmcilroy29, I didn't integrate any changes, because : 

https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504#discussion_r463441342 raises the question about moving this in Device Object directly.

I also think this would be better (https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504#discussion_r463470429). but this means we need to create a v1.2 for Device object.
The issue is that we need this object version available for lwm2m 1.0 and Device object v1.1 already target lwm2m 1.1.

My guess is that this DeviceExt object is needed at least for lwm2m 1.0, but I prefer to wait for OMA feedback/confirmation before to resolve observation.

**Comment by sbernard31** [comment]
@seanmcilroy29 , @dnav , @akeranen any news about my concern :point_up:  (https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504#issuecomment-677443802)

**Comment by akeranen** [comment]
This is an interesting question and shows some of the problems of supposedly-linear versioning. If we had features instead of versions of the protocol, you could use all versions of Device object with all versions of the protocol and just not use the features not supported. Something to think about for v2.0. Meanwhile, if you really need these to work with 1.0, new object seems like the best approach.

**Comment by sbernard31** [comment]
@akeranen, thx for the answer. I will apply requested changes to this PR now.

**Comment by sbernard31** [comment]
I hope I take all the comments into account.
I did it in a separated commit to make review of new changes easier, but I can squash all commit in one if needed.

**Comment by seanmcilroy29** [comment]
@sbernard31 could you squash all your commits into this PR.
Tomorrow IPSO will be holding a Conf Call and this Object will be submitted for Approval

**Comment by sbernard31** [comment]
@seanmcilroy29  done. :)

**Comment by jpradocueva** [comment]
Since this Pull Request was created the "test" branch has undergone major changes.
Once there is an agreement with this Object. The maintainer will move the Object directly to the new "test" branch.

**Comment by jpradocueva** [comment]
The name of the ObjectID was updated in the DDF.xml to match the name of the Object "Reboot Status". 
Now it passes validation.

**Comment by mojanm** [comment]
Having read through https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/280 that has triggered this PR, the proposed object certainly seems to address the issue and looks good. The only concern I have is doing exec commands on resources of device object suddenly triggers changes to this object without any relationship being established between the two objects which in my view is something of a departure for LwM2M? I fully understand the desire to avoid making significant changes to device object at this stage, but maybe it would be possible to add executable resources for Factory reset and Reboot to this object so that we have a self contained reboot/factory reset object and for the clients that support this object the server can control reboot and factory reset via this object entirely? Factory reset is an optional resource in the device object, and its duplication can be avoided. However, reboot is mandatory which means we have a duplicate resource in two objects, but to me that we can live with.

File: 10371.xml
Comment (jaimejim) [inline 10371.xml:33]: "Reboot Status" sounds good to me.

File: 10371.xml
Comment (jaimejim) [inline 10371.xml:73]: Error message is an unformatted string, with a dump of the error. I guess in the long run we might want to have something more specific about how the error is formatted. That's for OMA to specify, not on this Object in particular.

File: 10371.xml
Comment (jaimejim) [inline 10371.xml:83]: As others have mentioned, instead of extending, it would be good to directly update /3 with these resources. Maybe DM can have a look and add it to the spec at some point?

**Review by jaimejim — APPROVED** [review]
I provided some comments on the review but all in all the current version is valid. I also checked with the validator and it passes validation (I forgot that we have that automated nowadays :D ) 

I also have the same comment as @mojanm that we are adding extension objects that trigger executable resources on DM objects might have unintended consequences. It would be better to extend DM objects instead.

**Comment by sbernard31** [comment]
As I said before, I agree that this makes probably more sense to put this is a device object but it sounds this is not possible to get it compatible for LWM2M 1.0 (see https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504#issuecomment-677443802 and https://github.com/OpenMobileAlliance/lwm2m-registry/pull/504#issuecomment-682666160)

About, duplicating executable resources from Device to this Object, I can not see the benefits of this.
I don't even sure what would that means concretely :thinking:. 
Reboot status would only be available when exec resource from this object is used but exec on device object will not change status of this object ? (meaning this is more complicate for devices to implement)
or both resource will do exactly the same thing and so I do not understand the what this is better.

By the way, if we do that the name and the documentation of this object is no more adapted.

**Comment by mkgillmore** [comment]
Approved via R&A pending validation

**Pull request merged**
